GameBoard/Controllers/FriendsController.cs

```diff
@@ -30,6 +33,15 @@ public class FriendsController : Controller
             _hostConfiguration = hostConfiguration.Value;
         }
 
+        [HttpGet]
+        public async Task<IActionResult> SearchFriendsForGroup(string input, string userName, string groupId)
+        {
+            var friends = await _friendsService.GetFriendsByUserNameAsync(userName);
+            friends = friends.Where(x => x.UserName.ToUpper().Contains(input.ToUpper()));
+            var model = new FriendSearchResultViewModel(friends.Select(u => u.ToViewModel()), groupId);
+            return ViewComponent("FriendSearchResults", model);
```

Comment:
You've put logic into a controller. Extract the friends filtering to a method, preferably in the `UserSearchService`.

Reply:
It requires friendsService, so could it be there instead?

Reply:
Done

Reply:
Not really done but it's 4 a.m.